src/drivers/intel/cavs/ipc.c

```diff
@@ -58,6 +58,9 @@ static void ipc_irq_handler(void *arg)
 {
 	struct ipc *ipc = arg;
 	uint32_t dipcctl;
+	uint32_t flags;
+
+	spin_lock_irq(&ipc->lock, flags);
```

Comment:
could you explain a bit more what reply from the host could be racing with secondary cores processing IPCs? Do you mean IPC messages from the DSP like trace updates?

Reply:
yes, see line 104 "/* reply message(done) from host */", the ack to FW->host notification message, e.g. trace updates.
That's why we always seeing trace update before the IPC timeout.